spec/units/emergency_banner_spec.rb

```diff
@@ -0,0 +1,55 @@
+RSpec.describe GovukWebBanners::EmergencyBanner do
+  let(:redis_client) { double(hgetall: {}) }
+  subject(:emergency_banner) { GovukWebBanners::EmergencyBanner.new(redis_client:) }
+
+  describe "caching" do
+    context "with a Rails cache" do
+      it "caches calls to the redis client for one minute" do
+        allow(Rails).to receive(:cache).and_return(ActiveSupport::Cache.lookup_store(:memory_store))
+        Rails.cache.clear
+
+        GovukWebBanners::EmergencyBanner.new(redis_client:)
+        GovukWebBanners::EmergencyBanner.new(redis_client:)
+
+        expect(redis_client).to have_received(:hgetall).once
+
+        travel_to(Time.now + 61.seconds)
+
+        GovukWebBanners::EmergencyBanner.new(redis_client:)
+        expect(redis_client).to have_received(:hgetall).twice
+
+        travel_back
+      end
+    end
+  end
+
+  context "with the emergency banner inactive" do
+    describe "#active?" do
+      it "returns false" do
```

Comment:
Minor: Is a separate context need for this?

To reduce the amount of nesting, could there not instead be a top-level `describe "#active?` with tests :
* `it "returns false when the emergency banner is inactive"`
* `it "returns true when the emergency banner is active"`
* `it "returns false if the call to Redis fails"`

Reply:
Ah yes - originally there were more methods to test here so this nesting made sense, but they got factored out. Have inverted the nesting.